src/Chapter3.hs

```diff
@@ -560,22 +598,30 @@ introducing extra newtypes.
 🕯 HINT: if you complete this task properly, you don't need to change the
     implementation of the "hitPlayer" function at all!
 -}
+newtype Hp = Hp Int
+newtype Att = Att Int
+newtype Def = Def Int
+newtype Dmg = Dmg Int
+newtype Arm = Arm Int
+newtype Dex = Dex Int
+newtype Str = Str Int
+
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
+    { playerHealth    :: Hp
+    , playerArmor     :: Arm
+    , playerAttack    :: Att
+    , playerDexterity :: Dex
+    , playerStrength  :: Str
     }
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
+calculatePlayerDamage :: Att -> Str -> Dmg
+calculatePlayerDamage (Att attack) (Str strength) = Dmg $ attack + strength
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDefense :: Arm -> Dex -> Def
+calculatePlayerDefense (Arm armor) (Dex dexterity) = Def $ armor * dexterity
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerHit :: Dmg -> Def -> Hp -> Hp
+calculatePlayerHit (Dmg damage) (Def defense) (Hp health) = Hp $ health + defense - damage
```

Comment:
Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype